env/plugin-commands-env/test/node.test.ts

```diff
@@ -84,3 +85,24 @@ test('get node version base dir', async () => {
   const versionDir = getNodeVersionsBaseDir(process.cwd())
   expect(versionDir).toBe(path.resolve(process.cwd(), 'nodejs'))
 })
+
+describe('prepareExecutionEnv', () => {
+  test('should not proceed to fetch Node.js if the process is already running in wanted node version', async () => {
+    const _tempDir = tempDir()
+
+    await prepareExecutionEnv({
+      bin: '',
+      pnpmHomeDir: process.cwd(),
+      rawConfig: {},
+    }, {
+      executionEnv: { nodeVersion: process.versions.node },
+    })
+
+    expect(fetchMock).not.toHaveBeenCalled()
+
+    // We retry tests once. The test always passes on retry, because requested
+    // node version exists, hence actually no call to download node
+    // so we need to clean up the temp dir, to make sure we are testing the right thing
+    rmSync(_tempDir, { recursive: true })
```

Comment:
why should it be removed if there will be a new random temporary directory on rerun?

Reply:
Yeah that's right. I wanted to prove to myself this test needs that rmDir, but there's something wrong. The test always passes even without my implementation in place. Looking into it and will be back to you :detective:

Reply:
I can't set retryTimes in describe block, because it leaks to other tests as well. For instance, with the change below, the `something` test I wrote, always passes, because of the retry policy. But if I add that `jest.retryTimes(0)` for my test, it also leaks and fails the `something` test.

![image](https://github.com/user-attachments/assets/1c9d6f49-3d1d-44a8-9822-1521f6b51d96)

In the other hand, the `nodeFetchPromises` is implementation details, I can export and empty it after each test run. 

![image](https://github.com/user-attachments/assets/acc9933d-0fc7-4551-98b8-4008b6c8a341)


But none of those solutions are ideal. Any idea?

Reply:
Yeah you're right. I changed the test to mock implementation of fetchMock to throw: https://github.com/pnpm/pnpm/pull/8673/commits/51e308ba37e1586b88de564d7b7c8217b6922d19

This way the `nodeFetchPromises` cache does not allow the test to pass on second retry.

Thanks

Reply:
Independent of whether deleting the directory is required for test correctness on reruns, I actually do prefer tests properly clean up the temporary directories they create. I think the current pattern in pnpm's tests is not to do that though.

Reply:
We can do it in a global teardown function but not in scope of this PR.

Reply:
> Independent of whether deleting the directory is required for test correctness on reruns, I actually do prefer tests properly clean up the temporary directories they create. I think the current pattern in pnpm's tests is not to do that though.

Yes, the comment in `jest.setup.js` also agrees, and apparently that's the whole philosophy of retrying tests once: to catch some of the tests/code that do not properly clean up.

However note that in the scope of this PR there are no cleanups required, because it does not download antything